src/command.rs

```diff
@@ -179,8 +176,56 @@ where
     ///     and can therefore not be written to the given `writer`.
     ///     Therefore, there is no difference between [execute](./trait.ExecutableCommand.html)
     ///     and [queue](./trait.QueueableCommand.html) for those old Windows versions.
-    fn execute(&mut self, command: impl Command<AnsiType = A>) -> Result<&mut Self> {
-        execute!(self, command)?;
+    fn execute(&mut self, command: impl Command) -> Result<&mut Self> {
+        self.queue(command)?;
+        self.flush()?;
         Ok(self)
     }
 }
+
+/// Writes the ANSI representation of a command to the given writer.
+fn write_command_ansi<C: Command>(
+    io: &mut (impl io::Write + ?Sized),
+    command: C,
+) -> io::Result<()> {
+    struct Adapter<T> {
```

Comment:
Im not sure why we need this adaptor? Can we not write directly to 'io'?

Reply:
The commands write to a `fmt::Write` type, whereas this is an `io::Write` type. The [standard library](https://github.com/rust-lang/rust/blob/master/library/std/src/io/mod.rs#L1499) has a similar adaptor. The alternative is to have an adaptor that wraps a `Command` and implements `Display`, however that has to create and go through a `Formatter`, while this writes it directly.

Reply:
Okey perfect